backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/view/OperationInterpreterViewSwitch.java

```diff
@@ -0,0 +1,206 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.components.emf.view;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.eclipse.emf.common.util.EList;
+import org.eclipse.emf.ecore.EClass;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.EPackage;
+import org.eclipse.emf.ecore.EStructuralFeature;
+import org.eclipse.emf.ecore.util.EcoreUtil;
+import org.eclipse.emf.edit.domain.EditingDomain;
+import org.eclipse.sirius.components.core.api.IEditService;
+import org.eclipse.sirius.components.core.api.IEditingContext;
+import org.eclipse.sirius.components.emf.services.EditingContext;
+import org.eclipse.sirius.components.interpreter.AQLInterpreter;
+import org.eclipse.sirius.components.interpreter.Result;
+import org.eclipse.sirius.components.representations.VariableManager;
+import org.eclipse.sirius.components.view.ChangeContext;
+import org.eclipse.sirius.components.view.CreateInstance;
+import org.eclipse.sirius.components.view.DeleteElement;
+import org.eclipse.sirius.components.view.Operation;
+import org.eclipse.sirius.components.view.SetValue;
+import org.eclipse.sirius.components.view.UnsetValue;
+import org.eclipse.sirius.components.view.util.ViewSwitch;
+import org.eclipse.sirius.ecore.extender.business.internal.accessor.ecore.EcoreIntrinsicExtender;
+
+/**
+ * This switch allows to handle common {@link Operation}s without dependencies to a specific representation.
+ *
+ * @author fbarbin
+ */
+public class OperationInterpreterViewSwitch extends ViewSwitch<Optional<VariableManager>> {
```

Comment:
I failed to see that you have used inheritance for that, I mentioned this week that you should use composition not inheritance for that. We had problems with the composition of business code in the past. This is a disguised abstract class that no one instantiate and which is used only to share business code with multiple implementation.

Reply:
You mean that the DiagramOperationInterpreterViewSwitch should delegate to the OperationInterpreterViewSwitch  (instanciated by it self) for common cases instead of inherit from it?